### PR TITLE
Update Name class and User Guide

### PR DESCRIPTION
**Pull request by darrentfy**

### CODE REVIEW

**Comment by darrentfy** [comment]
Updated Name class to have a character limit of 100, updated test cases accordingly. 
Removed extra white spaces within names.
Updated User Guide to include the new requirements, and added invalid examples for commands

File: docs/UserGuide.md [hunk 1]
```diff
@@ -79,18 +79,26 @@ Format: `add n/STUDENT_NAME c/CONTACT_NUMBER a/HOME_ADDRESS [r/RISK_LEVEL]`
...
    - Must be `high`, `medium`, or `low`
 
 Examples:
 * `add n/John c/81349705 a/Yishun Street 56 Blk 21 #05-07`
 * `add n/Sally c/94149785 a/Woodlands Street 11 Blk 888 #08-08 r/low`
 
+Invalid examples:
+* `add n/Sally c/94149785` (missing address)
+* `add n/Sa11y! c/1234 a/Woodlands Street 11 Blk 888 #08-08` (invalid name and phone number)
+* `add n/Sa11y! c/94149785 a/` (address is blank)
```
Comment (SimWPEric) [inline docs/UserGuide.md:99]: Maybe `Sally` instead of `Sa11y` is better here because the error is address only.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -79,18 +79,26 @@ Format: `add n/STUDENT_NAME c/CONTACT_NUMBER a/HOME_ADDRESS [r/RISK_LEVEL]`
...
+* `add n/Sa11y! c/1234 a/Woodlands Street 11 Blk 888 #08-08 r/abc` (invalid risk level)
```
Comment (SimWPEric) [inline docs/UserGuide.md:100]: Same for this

**Review by SimWPEric — CHANGES_REQUESTED** [review]
Very minor fix needed for the `Sa11y` part. Invalid examples are clear and easy to understand.

File: docs/UserGuide.md (hunk 2 above)
Comment (darrentfy) [reply docs/UserGuide.md]: Great point, updated the relevant examples

**Review by SimWPEric — APPROVED** [review]
LGTM

**Pull request merged**